dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -766,6 +826,22 @@ void ExchangeReceiverBase<RPCContext>::connectionDone(
         finishAllMsgChannels();
 }
 
+template <typename RPCContext>
+void ExchangeReceiverBase<RPCContext>::connectionLocalDone()
+{
+    std::lock_guard lock(mu);
+    is_local_conn_alive = false;
+    cv.notify_all();
+}
+
+template <typename RPCContext>
+void ExchangeReceiverBase<RPCContext>::setLocalAlive()
+{
+    std::lock_guard lock(mu);
+    RUNTIME_CHECK_MSG(!is_local_conn_alive, "is_local_conn_alive should only be set once");
```

Comment:
```suggestion
    RUNTIME_ASSERT(!is_local_conn_alive, "is_local_conn_alive should only be set once");
```
Because we have no way to recover here.